nntrainer/graph/network_graph.h

```diff
@@ -32,6 +32,9 @@ namespace nntrainer {
 struct LayerNode {
   std::shared_ptr<Layer> layer;
   unsigned int index;
+#ifdef PROFILE
+  unsigned int event_key;
```

Comment:
how about the event to be either of string or int?
In that way, the layer name can be the unique event id.

Reply:
Implementation wise, the vector could be replaced with unordered_map/map in profiler.

Reply:
To minimize the profiler bottleneck, I will use int as a universal key for now.
Let's change the string key later when it is needed so that we can measure the impact for the change.